kbase-extension/static/kbase/js/util/icon.js

```diff
@@ -1,130 +1,181 @@
+
+        const style = color ? {color: color} : {};
+        const layers = stacked ? _makeStackLayers(shape, color) : [];
+        iconClass = iconClass.replace('icon ', '');
+
+        return span({
+            class: `${cssBaseName}__container--${iconType} fa-stack`,
+        }, layers.concat(
+            span({
+                class: `${cssBaseName}__icon_background--${iconType} fa fa-${shape} fa-stack-2x`,
+                style: style,
+            }),
+            span({
+                class: `${cssBaseName}__icon--${iconType} fa fa-inverse fa-stack-1x ${iconClass}`,
+            })
+        ));
+    }
+
+    /**
+     * @method
+     * @public
+     * Builds a data icon for a given data type
+     * @param {string} type The shortened type of object to fetch the icon for. E.g. "Genome" not
+     * "KBaseGenomes.Genome-2.1"
+     * @param {boolean} stacked If true, creates a stacked effect of multiple 'icons'.
+     */
+    function makeDataIcon(type, stacked) {
+        const iconType = stacked ? 'data-stack' : 'data',
+            color = _.has(iconSpec.color_mapping, type)
+                ? iconSpec.color_mapping[type]
+                : logoColorLookup(type),
+            icon = _.has(iconSpec.data, type)
+                ? iconSpec.data[type]
+                : iconSpec.data.DEFAULT;
+        return _makeIcon(iconType, icon[0], 'circle', color, stacked);
+    }
+
+    /**
+     * @method
+     * @public
+     * Builds a data icon in the given $logo div.
+     * @param {object} $logo jQuery element in which to insert an icon.
+     * @param {string} type The shortened type of object to fetch the icon for. E.g. "Genome" not
+     * "KBaseGenomes.Genome-2.1"
+     * @param {boolean} stacked If true, creates a stacked effect of multiple 'icons'.
+     * @param {int} indent -- deprecated
+     */
+    function buildDataIcon($logo, type, stacked, indent) {
+        const iconStr = makeDataIcon(type, stacked);
+        $logo.append(iconStr);
         return $logo;
     }
 
     /**
-     * Whether the stack of icons is using font-awesome
-     * or our own custom set.
-     *
-     * @param {list of str} iconList  Icon classes, from icons.json
-     * @returns {boolean}
+     * Legacy code that overwrites any existing DOM elements in params.elt. Used for passing in
+     * a set of parameters to makeDataIcon as an object instead of a list.
+     * @param {object} params - The set of parameters to use.
+     * @param {object} params.elt - The JQuery element containing the logo to overwrite.
+     * @param {string} params.type - The type of object to build a logo from.
+     * @param {boolean} params.stacked - If true, creates a stacked effect of multiple icons on each other
      */
-    function isCustomIcon(iconList) {
-        return (iconList.length > 0 && iconList[0].length > 4 &&
-            iconList[0].substring(0, 4) == 'icon');
+    function overwriteDataIcon(params) {
```

Comment:
I couldn't find this method anywhere -- is there any point in keeping it?

Reply:
Nope. I'm sure the module that used it was deleted in some other cleanup effort. Go ahead and delete.